code/modules/dwarven/dwarven_structures.dm

```diff
@@ -0,0 +1,428 @@
+/obj/structure/destructible/dwarven
+	density = TRUE
+	anchored = TRUE
+	icon = 'icons/obj/dwarven.dmi'
+	light_power = 1
+	var/cooldowntime = 0
+	custom_materials = list()
+	//break_sound = 'sound/hallucinations/veryfar_noise.ogg' //to do: find a suitable noise
+
+/obj/structure/destructible/dwarven/New()
+	START_PROCESSING(SSfastprocess, src)
+	..()
+
+/obj/structure/destructible/dwarven/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus
+	name = "Ancient Sarcophagus"
+	icon_state = "sarcophagus_open"
+
+	var/recharge = TRUE
+	var/recharge_points = 0
+	var/recharge_points_max = 500
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>It is [(recharge_points*100)/recharge_points_max]% powered </span>"
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus/attackby(obj/item/stack/ore/I, mob/living/user, params)
+	recharge_points += I.amount * I.points
+	qdel(I)
+	try_to_activate()
+	return
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus/proc/try_to_activate()
+	if(recharge_points_max <= recharge_points)
+		for(var/mob/M in viewers(src,5))
+			to_chat(M, "<span class='notice'>The sarcophagus reignites with ancient fire, ready to birth another dwarf!</span>")
+			new  /obj/effect/mob_spawn/human/dwarven_sarcophagus(get_turf(src))
+			qdel(src)
+	else
+		for(var/mob/M in viewers(src,5))
+			to_chat(M, "<span class='notice'>The sarcophagus is set ablaze for a second, but ancient powers die down. It requires more minerals!</span>")
+
+/obj/structure/destructible/dwarven/lava_forge
+	name = "Ancient Forge"
+	icon_state = "dwarven_forge"
+	custom_materials = list(/datum/material/iron = 20000)
+
+/obj/structure/destructible/dwarven/lava_forge/attackby(obj/item/I, mob/living/user, params)
+	var/mob/living/carbon/human/H = user
+	if(!H.has_language(/datum/language/dwarven))
+		to_chat(H, "<span class='notice'>You don't understand how to operate this ancient machinery!</span>")
+		return
+	if(!istype(I, /obj/item/stack/ore))
+		return ..()
+	var/obj/item/stack/ore/O = I
+	var/obj/item/refined = O.refined_type
+	if(isnull(refined))
+		return
+	try_forge(user, O, refined)
+
+/obj/structure/destructible/dwarven/lava_forge/proc/try_forge(mob/user, obj/item/stack/ore/O, obj/item/refined_result, recursive = FALSE)
+	if(!recursive) //Only say this the first time
+		to_chat(user, "<span class='notice'>You start smelting [O] into [initial(refined_result.name)].</span>")
+	if(!do_after(user, 30, target = src))
```

Comment:
you can spam this to blast through entire stacks, you could also tire this to operating

Reply:
intentional